q2_types_genomics/genome_data/_transformer.py

```diff
@@ -0,0 +1,97 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2021, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import collections.abc
+import glob
+import os
+
+import pandas as pd
+import skbio
+from q2_types.feature_data._transformer import _fastaformats_to_series
+from skbio.io import read
+
+from . import (
+    GenesDirectoryFormat, ProteinsDirectoryFormat, GFF3Format
+)
+from ..plugin_setup import plugin
+
+CONSTRUCTORS = {
+    'DNA': skbio.DNA,
+    'RNA': skbio.RNA,
+    'protein': skbio.Protein
+}
+
+
+def _series_to_fasta(series, ff, seq_type='DNA'):
+    fp = os.path.join(ff.path, f'{series.name}.fasta')
+    with open(fp, 'w') as fh:
+        for id_, seq in series.iteritems():
+            if seq:
+                sequence = CONSTRUCTORS[seq_type](seq, metadata={'id': id_})
+                skbio.io.write(sequence, format='fasta', into=fh)
+
+
+@plugin.register_transformer
+def _1(dirfmt: GenesDirectoryFormat) -> pd.DataFrame:
+    data = {}
+    for fp in sorted(glob.glob(os.path.join(str(dirfmt), '*.fa*'))):
```

Comment:
I missed this, but you can "chain" transformers by accessing the FileCollections directly (when this is updated to be a standard DirFmt). So you would be able to say something like `dirfmt.genes.iter_views(pd.DataFrame)` and then merge/concat them together.

Reply:
Oh, that's neat - I missed that! I updated the genes and proteins dirfmt transformers to use that - hope it makes more sense now.